src/client.ts

```diff
@@ -196,7 +275,11 @@ export default class Client {
           await this.ruby.activateRuby();
         }
 
-        await this.restart();
+        if (this.client?.isRunning()) {
+          await this.restart();
+        } else {
+          await this.start();
+        }
```

Comment:
I usually find this logic being encapsulated in the behaviour of `restart` to produce mush more readable code. If we make `restart` to the right thing based on if the client is running or not, then callers can just call `restart` when the intent is to do that.